### ISSUE TITLE
big difference after converting model to pytorch

### ISSUE THREAD

**Issue by noahzn** [rpautrat/SuperPoint#316]
Hi, I use the `convert_to_pytorch` code to convert my TF model. The results are very different.
1. TF version
![image](https://github.com/rpautrat/SuperPoint/assets/8244850/82e931ca-8ff7-4739-8d4f-b6a650e525ae)
2. Pytorch version
![image](https://github.com/rpautrat/SuperPoint/assets/8244850/94d130a0-b7f3-4596-9f77-49c19af8cc57)

I used the default parameters. Could you help me with that? Thank you. @rpautrat @sarlinpe

**Comment by sarlinpe** [comment]
The TF output looks very different from the one in the `convert_to_pytorch.ipynb` notebook - did you change any parameter there? If not, this is very surprising. I used `tensorflow==1.15.0` and `torch==1.13.1+cu117`, what versions are you using? You might need to compare the results layer by layer to find out where the discrepancy comes from - I recommend starting from the raw score map.

**Comment by sarlinpe** [comment]
I'm using Python 3.7. What is reported in the cell checking the different of dense outputs?
```
Diff logits: 3.1471252e-05 4.6826185e-06 3.874302e-06 max/mean/median
Diff descriptors: 2.041459e-06 2.7050896e-07 2.2351742e-07 max/mean/median
```
You could maybe increase the detection threshold for the PyTorch model, but this would not solve the underlying issue - there must be an implementation difference somewhere.

**Comment by noahzn** [comment]
> I'm using Python 3.7. What is reported in the cell checking the different of dense outputs?
> 
> ```
> Diff logits: 3.1471252e-05 4.6826185e-06 3.874302e-06 max/mean/median
> Diff descriptors: 2.041459e-06 2.7050896e-07 2.2351742e-07 max/mean/median
> ```
> 
> You could maybe increase the detection threshold for the PyTorch model, but this would not solve the underlying issue - there must be an implementation difference somewhere.

Diff logits: 3.4570694e-05 3.1457146e-06 2.6226044e-06 max/mean/median
Diff descriptors: 7.6293945e-06 2.6236998e-07 2.0861626e-07 max/mean/median

**Comment by sarlinpe** [comment]
The diff numbers are of the same order of magnitude. The issue must be in the keypoint selection, somewhere in this section:
https://github.com/rpautrat/SuperPoint/blob/d8ebb9040fac489e23dd0b6f136976c329eed3ba/superpoint_pytorch.py#L135-L157
Do you mind sharing your checkpoint?

**Comment by noahzn** [comment]
[model.zip](https://github.com/rpautrat/SuperPoint/files/14436310/model.zip)


> The diff numbers are of the same order of magnitude. The issue must be in the keypoint selection, somewhere in this section:
> 
> https://github.com/rpautrat/SuperPoint/blob/d8ebb9040fac489e23dd0b6f136976c329eed3ba/superpoint_pytorch.py#L135-L157
> 
> 
> Do you mind sharing your checkpoint?

Here is my checkpoint.

**Comment by sarlinpe** [comment]
Thanks. I found the issue and fixed it in PR https://github.com/rpautrat/SuperPoint/pull/317. I suggest increasing your detection threshold to 0.01, the results will look closer to the model we have trained.

**Comment by noahzn** [comment]
> Thanks. I found the issue and fixed it in PR #317. I suggest increasing your detection threshold to 0.01, the results will look closer to the model we have trained.

Thank you very much for the fix! Now the results look closer! Since I am going to use this trained model with LightGlue, I am wondering if glue-factory's open SuperPoint code has a similar nms problem??

**Comment by sarlinpe** [comment]
No, your converted model and glue-factory's open SP are fine - the problem was in the inference TensorFlow model
